File: app/src/main/java/com/example/groundzero/compat/CompatibilityModels.kt
package com.example.groundzero.compat

data class DomainProfile(
    val raw: Double,
    val pct: Int,
    val bucket: String,
)

data class GZProfile(
    val domains: Map<String, DomainProfile>,
    /** Keys like "O:Imagination", values 1–5 raw. */
    val facets: Map<String, Double>,
    val archetypeId: String,
)

data class DomainCompatEntry(
    val aRaw: Double,
    val bRaw: Double,
    val delta: Double,
    val synergy: String,
    val scorePct: Double,
)

data class FacetAlignPair(
    val facet: String,
    val a: Double,
    val b: Double,
)

data class FacetConflictPair(
    val facet: String,
    val a: Double,
    val b: Double,
)

data class FacetAnalysis(
    val alignPairs: List<FacetAlignPair>,
    val conflictPairs: List<FacetConflictPair>,
)

data class OverallCompat(
    val scorePct: Int,
    val band: String,
    val rationale: List<String>,
)

data class CompatResult(
    val overall: OverallCompat,
    val domains: Map<String, DomainCompatEntry>,
    val facets: FacetAnalysis,
)

data class OverrideItem(val id: String, val why: String)
data class RoutineItem(val name: String, val spec: String)
data class PrescriptionScenarios(val work: List<String>, val relationship: List<String>)

data class Prescriptions(
    val overrides: List<OverrideItem>,
    val routines: List<RoutineItem>,
    val scenarios: PrescriptionScenarios,
)

data class CompatibilityBundle(
    val compat: CompatResult,
    val prescriptions: Prescriptions,
    /** 2–4 sentence human-readable narrative of the overall dynamic. */
    val narrative: String,
    /** How the two archetypes tend to combine — deterministic from sorted archetype pair. */
    val archetypePairing: String,
)

File: iphobe app/GroundZeroiOS/Core/Models/CompatibilityModels.swift
import Foundation

struct DomainProfile: Codable {
    let raw: Double
    let pct: Int
    let bucket: String
}

struct GZProfile: Codable {
    let domains: [String: DomainProfile]
    let facets: [String: Double]
    let archetypeId: String
}

struct DomainCompatEntry: Codable {
    let aRaw: Double
    let bRaw: Double
    let delta: Double
    let synergy: String
    let scorePct: Double
}

struct FacetAlignPair: Codable {
    let facet: String
    let a: Double
    let b: Double
}

struct FacetConflictPair: Codable {
    let facet: String
    let a: Double
    let b: Double
}

struct FacetAnalysis: Codable {
    let alignPairs: [FacetAlignPair]
    let conflictPairs: [FacetConflictPair]
}

struct OverallCompat: Codable {
    let scorePct: Int
    let band: String
    let rationale: [String]
}

struct CompatResult: Codable {
    let overall: OverallCompat
    let domains: [String: DomainCompatEntry]
    let facets: FacetAnalysis
}

struct OverrideItem: Codable, Hashable {
    let id: String
    let why: String
}

struct RoutineItem: Codable, Hashable {
    let name: String
    let spec: String
}

struct PrescriptionScenarios: Codable {
    let work: [String]
    let relationship: [String]
}

struct Prescriptions: Codable {
    let overrides: [OverrideItem]
    let routines: [RoutineItem]
    let scenarios: PrescriptionScenarios
}

struct CompatibilityBundle: Codable {
    let compat: CompatResult
    let prescriptions: Prescriptions
    let narrative: String
    let archetypePairing: String
}
